ee_tests/src/page_objects/user_profile.page.ts

```diff
@@ -59,26 +60,63 @@ export class CleanupUserEnvPage extends AppPage {
 
   async cleanup(username: string) {
     await this.eraseEnvButton.clickWhenReady();
-
     let confirmationElement =  this.innerElement(ui.BaseElement, 'modal', '');
     let confirmationBox = new CleanupConfirmationModal(confirmationElement);
 
     await confirmationBox.ready();
+
     await confirmationBox.confirmationInput.enterText(username);
     await confirmationBox.confirmEraseButton.clickWhenReady();
-    await browser.wait(until.presenceOf(
+
+    await browser.wait(until.presenceOf(this.alertBox), support.LONG_WAIT);
+    let alertText = await this.alertBox.getText();
+    support.info('Alert text = ' + alertText);
+
+    /* An intermittent error is resulting in some user enviroment resets failing
+       https://github.com/openshiftio/openshift.io/issues/1637
+       If this happens - try one more time - if that fails something serious
+       has happened and the test will/and should fail */
+
+    if (alertText.includes('Your OpenShift.io environment has been erased')) {
+      await browser.wait(until.presenceOf(
+      element(by.cssContainingText('fabric8-cleanup', 'Your OpenShift.io environment has been erased!'))));
+      support.info('***Env reset SUCCESS');
+    } else {
+      support.info('***Env reset FAILURE');
+
+      /* Retry - once - to reset env */
+      await this.dashboardButton.clickWhenReady();
+      let dashboardPage = new MainDashboardPage();
+      let userProfilePage = await dashboardPage.gotoUserProfile();
+      let editProfilePage = await userProfilePage.gotoEditProfile();
+      let cleanupEnvPage = await editProfilePage.gotoResetEnvironment();
+
+      await this.eraseEnvButton.clickWhenReady();
+      let confirmationElement1 =  this.innerElement(ui.BaseElement, 'modal', '');
+      let confirmationBox1 = new CleanupConfirmationModal(confirmationElement1);
+
+      await confirmationBox1.ready();
+      await confirmationBox1.confirmationInput.enterText(username);
+      await confirmationBox1.confirmEraseButton.clickWhenReady();
```

Comment:
Are you sure that navigating back to the reset page is necessary? Won't it stay on the same page? If it is necessary, refactore common code to sa separate function.